dandi/metadata.py

```diff
@@ -71,6 +91,19 @@ def get_metadata(path: Union[str, Path]) -> Optional[dict]:
             lgr.debug("Failed to get metadata for %s: %s", path, exc)
             return None
 
+    # Pretty fragile way to determine that it's not nwb.
+    # Ideally can find something better, like `is_nwb` or `is_bids`.
+    try:
+        h5py.File(path)
```

Comment:
I do not like this one bit.  How would a human looking at a dataset know whether something is BIDS or not?

Reply:
also bids datasets themselves can have nwb files. 

i would say bids datasets in dandi would still have dandiset.yaml. and that the check would be that each bids dataset in a dandiset will have a dataset_description.json.

algorithm for is_bids:
- walk up from file to find dataset_description.json and a dandiset.yaml 

if dandiset.yaml occurs before a dataset_description then the dataset is not bids. 

also for this PR, which is for extracting metadata, it seems the simplest check would be, is there a same named sidecar json file?

this also means when dandi cli downloads a bids dandiset, it has to download the relevant dataset_description.json (there are multiple). this is to support partial uploads for large dandisets that are bids. the alternative would be to break up all bids datasets whether "raw" or "derived" into separate dandisets, and at least for the ones we have now, it would make that ugly, and we need a way to link better through provenance.

@TheChymera - would be happy to chat briefly about this. i think in the first pass this can be simplified quite a bit, and all it may require is simply parsing the file name into sub, session, sample, stain, modality.

Reply:
@jwodder improved detection as per @satra 's suggestion. Also made the path creation more windows-robust (though I think it *might* have worked anyway, we noticed in some other PRs that `os.path` automagically windows-ized things on windows).